File: app/src/main/java/com/damhoe/scoresheetskat/score/domain/Score.java
package com.damhoe.scoresheetskat.score.domain;

public abstract class Score {
   protected long id;
   protected int playerPosition;
   protected long gameId;
   public abstract int toPoints();

   public long getGameId() {
      return gameId;
   }

   public void setGameId(long gameId) {
      this.gameId = gameId;
   }

   public int getPlayerPosition() {
      return playerPosition;
   }

   public long getId() {
      return id;
   }

   public void setId(long id) {
      this.id = id;
   }

}

File: app/src/main/java/com/damhoe/skatscores/game/domain/score/Score.kt
package com.damhoe.skatscores.game.domain.score

import java.util.UUID

abstract class Score(
    var playerPosition: Int = 0,
    var gameId: Long = -1L
) {
    open var id: Long = UUID.randomUUID().mostSignificantBits
    abstract fun toPoints(): Int
}